Multisig/Cross-layer/Synchronization/SynchronizationService.swift

```diff
@@ -0,0 +1,43 @@
+//
+//  SynchronizationService.swift
+//  Multisig
+//
+//  Created by Andrey Scherbovich on 04.02.21.
+//  Copyright © 2021 Gnosis Ltd. All rights reserved.
+//
+
+import Foundation
+
+class SynchronizationService {
+    static let shared = SynchronizationService()
+
+    private var syncLoopRepeater: Repeater?
+    private let syncInterval: TimeInterval
+
+    init(syncInterval: TimeInterval = 5) {
+        self.syncInterval = syncInterval
+    }
+
+    /// Starts synchronisation loop on a background thread. Every `syncInterval` seconds the loop executes
+    /// If inbetween of these udpates the synchronisation is stopped, then all further actions are skipped.
+    func startSyncLoop() {
+        guard syncLoopRepeater == nil else { return }
+        DispatchQueue.global().async { [unowned self] in
+            // repeat syncronization loop every `syncInterval`
+            self.syncLoopRepeater = Repeater(delay: self.syncInterval) { repeater in
+                if repeater.isStopped { return }
+                WalletConnectServerController.shared.updatePendingTransactions()
```

Comment:
in the future we would need some sort of time to live for a pending transactions to discard the old ones

Reply:
I added ttl logic in this pr